examples/seismic/viscoelastic/operators.py

```diff
@@ -0,0 +1,208 @@
+import numpy as np
+import sympy as sp
+
+from devito import Eq, Operator, TimeFunction, NODE
+from examples.seismic.elastic import stress_fields, particle_velocity_fields, src_rec
+
+
+def memory_variables(model, save, space_order):
```

Comment:
This looks like it's the exact same setup as the stress tensor. Maybe a single function as tensor_function(name,...)?

Will be sing TensorFunction after my PR anyway but would be less work to update :D

Reply:
Sure, can do.